mri/optimizers/utils/cost.py

```diff
@@ -133,5 +142,8 @@ def _calc_cost(self, x_new, *args, **kwargs):
         cost: float
             the cost function defined by the operators (gradient + prox_op).
         """
-        cost = self.gradient_op.cost(x_new) + self.prox_op.cost(x_new)
+        if self.optimizer_type == 'forward_backward':
+            cost = self.gradient_op.cost(x_new) + self.prox_op.cost(x_new)
+        else:
+            cost = self.gradient_op.cost(x_new) + self.prox_op.cost(args[0])
```

Comment:
are we sure that `args[0]` is always going to be the linear op applied to the optimisation variable ?

in any case it's not very clear hear and deserves at least a comment, but I would be happy if there was a better way.

Reply:
I agree that this is bad and non-scalable code, but for now I could not find a better way to do it. 
I have added a TODO to generalize the code here, also added a comment explaining the code better

Reply:
ok sure